plugins/ckan_operators/package_operator.py

```diff
@@ -31,14 +31,32 @@ def __init__(self, address, apikey, **kwargs):
         self.ckan = ckanapi.RemoteCKAN(apikey=apikey, address=address)
 
     def execute(self, context):
-        packages = self.ckan.action.package_list()
+        # Initiate a loop of getting each package from CKAN, 1000 packages at a time
+        output = []
+        offset = 0
 
-        return {"packages": self.ckan.action.package_search(rows=len(packages))["results"]}
```

Comment:
Hey @careduz - I've replaced the 1000 hardcoded max packages to be returned with a smarter loop

On a loop:
It reads the count of packages returned after asking for 1000 and, if theres reason to believe there are more packages, it adds an offset and gets more packages

Reply:
Great, thank you!